rm/rm_ha.c

```diff
@@ -416,8 +416,12 @@ M0_INTERNAL void m0_rm_ha_unsubscribe(struct m0_rm_ha_tracker *tracker)
 {
 	struct m0_conf_obj *obj;
 
-	if (m0_clink_is_armed(&tracker->rht_conf_exp))
-		m0_clink_cleanup_locked(&tracker->rht_conf_exp);
+	if (m0_clink_is_armed(&tracker->rht_conf_exp)) {
+		if (m0_chan_is_locked(tracker->rht_conf_exp.cl_chan))
+			m0_clink_cleanup_locked(&tracker->rht_conf_exp);
+		else
+			m0_clink_cleanup(&tracker->rht_conf_exp);
+	}
```

Comment:
A situation with data structure can be either locked or unlocked is unusual. Please document why and when is this possible.